### PR TITLE
New bypass mode - http_versions

### PR DESCRIPTION
**Pull request by jtof-fap**
Added support for a new bypass mode -m http_versions.

### CODE REVIEW

**Comment by jtof-fap** [comment]
Diff of `bup-payloads-2022-08-19.lst` file with previous version:

```bash
diff --git a/tests-history/bup-payloads-2022-08-12.lst b/tests-history/bup-payloads-2022-08-19.lst
index 4da0488..3179c06 100644
--- a/tests-history/bup-payloads-2022-08-12.lst
+++ b/tests-history/bup-payloads-2022-08-19.lst
@@ -1,5 +1,5 @@

-Bypasser has generated 1781 payloads for 'http://127.0.0.1:8000/foo/bar' url:
+Bypasser has generated 1786 payloads for 'http://127.0.0.1:8000/foo/bar' url:
 [case_substitution] http://127.0.0.1:8000/Foo/bar
 [case_substitution] http://127.0.0.1:8000/fOo/bar
 [case_substitution] http://127.0.0.1:8000/foO/bar
@@ -825,6 +825,11 @@ Bypasser has generated 1781 payloads for 'http://127.0.0.1:8000/foo/bar' url:
 [http_methods] -X TRACE http://127.0.0.1:8000/foo/bar
 [http_methods] -X TRACK http://127.0.0.1:8000/foo/bar
 [http_methods] -X UPDATE http://127.0.0.1:8000/foo/bar
+[http_versions] --http0.9 http://127.0.0.1:8000/foo/bar
+[http_versions] --http1.0 http://127.0.0.1:8000/foo/bar
+[http_versions] --http1.1 http://127.0.0.1:8000/foo/bar
+[http_versions] --http2 http://127.0.0.1:8000/foo/bar
+[http_versions] --http2-prior-knowledge http://127.0.0.1:8000/foo/bar
 [mid_paths] http://127.0.0.1:8000/#?foo/bar
 [mid_paths] http://127.0.0.1:8000/#foo/bar
 [mid_paths] http://127.0.0.1:8000/%09%3bfoo/bar
```

File: bypass_url_parser.py [hunk 1]
```diff
@@ -408,11 +399,11 @@ def _run_curl(self, item):
                     result = result.replace(os.linesep, "\n")
 
                     # Store command result in CurlItem object
-                    if not self.proxy:
-                        item.response_raw_output = result
-                    else:
+                    if self.proxy and "HTTP/1.0 200 Connection established" in result:
```
Comment (laluka) [inline bypass_url_parser.py:402]: I'm not sure we want to hide this, a difference in the response line would also mean that the server works as an outbound proxy, which is a really interesting information !

Is this header from the `-x` curl option or for the test of the `CONNECT` header? 
I'm not sure, but it's worth keeping when we use the CONNECT or similar header, to detect working prroxies! ^.^

**Review by laluka — APPROVED** [review]
One small tweak requested, and maybe we should consider adding a changelog, but apart of that, good for merge ! :)

File: bypass_url_parser.py (hunk 1 above)
Comment (jtof-fap) [reply bypass_url_parser.py:402]: Noted point but no link with the `CONNECT` method or proxy detection. This fix resolves a bug in response parsing when a proxy is set (`-x http://127.0.0.1:8080`)

**Pull request merged**